packages/server/src/stages/rewrite-imports/index.ts

```diff
@@ -0,0 +1,85 @@
+import {Stage, transform} from "@blitzjs/file-pipeline"
+import path from "path"
+
+const isJavaScriptFile = (filepath: string) => filepath.match(/\.(ts|tsx|js|jsx)$/)
+
+const isInAppFolder = (s: string, cwd: string) => s.replace(cwd + path.sep, "").indexOf("app") === 0
+
+/**
+ * Returns a Stage that converts relative files paths to absolute
+ */
+export const createStageRewriteImports: Stage = ({config: {cwd}}) => {
+  const stream = transform.file((file) => {
+    const filecontents = file.contents
+    const filepath = file.path
+
+    if (!isJavaScriptFile(filepath) || !isInAppFolder(filepath, cwd) || filecontents === null) {
+      return file
+    }
+
+    const contents = filecontents.toString()
+
+    const newContents = replaceImports(contents)
+    file.contents = Buffer.from(newContents)
+
+    return file
+  })
+
+  return {stream}
+}
+
+export const patternImport = /(import\s+)(?:(?:(\*\s+as\s+\w+)|(\w+\s*)?(,\s*)?(?:(\{\s*\w+\s*(?:,\s*\w+\s*)*\}))?))(\s+from\s+["'])?([\w\.\\\/]+)(["']\)?)/gs
+
+export function replaceImports(content: string) {
+  return content.replace(patternImport, (...args) => {
+    const [
+      ,
+      ,
+      starImport,
+      defaultImportName,
+      combinedComma,
+      namedImportNames,
+      ,
+      origin,
+      ,
+    ] = args as string[]
+
+    const parts = origin.split("/")
+
+    if (parts.includes("pages")) {
+      if (parts[0] === "app") {
+        parts.splice(0, 1)
+      }
+    }
+
+    if (parts.includes("api")) {
+      parts.splice(0, parts.indexOf("api"), "pages")
+    }
+
+    if (starImport) {
+      // I don't know yet what to do
+      return args[0]
+    }
+
+    if (parts.includes("queries") || parts.includes("mutations")) {
```

Comment:
Now the juicy part, which deals with #734. Whenever we see smth like `import { myFunc } from "app/queries/myQuery"`, we rewrite that to `app/_resolvers/queries/myQuery`.

When we see smth like `import myQuery, { myFunc } from "app/queries/myQuery"`, we rewrite it to the following two:
```ts
import myQuery from "app/queries/myQuery"
import { myFunc } from "app/_resolvers/queries/myQuery"
```

Reply:
I've also tried having the generated `queries/myQuery` file export everything from the resolvers: https://github.com/blitz-js/blitz/pull/1456/commits/aa67b5a3679dec03f510d2cf14be5b24b7b4c42a

This has the downside that it also bundles all transitive dependencies of the query, a lot of which cannot be tree-shaken because they have side-effects on the module level (even our default `auth-utils.ts` since it initialises `secure-password` on the top level).

One possible solution would be to make the additional export in https://github.com/blitz-js/blitz/pull/1456/commits/aa67b5a3679dec03f510d2cf14be5b24b7b4c42a only appear when there's actually some other file importing from there, but that would required coordination with the RPC stage.